futures-util/src/lock/bilock.rs

```diff
@@ -69,6 +69,11 @@ impl<T> BiLock<T> {
         (Self { arc: arc.clone() }, Self { arc })
     }
 
+    /// Returns `true` only if the two `BiLock<T>`s originated from the same call to `BiLock::new`.
+    pub fn are_paired(first: &Self, second: &Self) -> bool {
```

Comment:
Why is this different than others? (are_paired(_, _) vs is_pair(&self, _)).

Reply:
There's no technical reason.  I went with the non-method implementation for BiLock because I felt like the instances are more independent than the sink/stream pair, but the primary purpose of the abstraction is to only be coupled to one other instance so the method syntax is still reasonable.  The other benefit of using the method signature is that it's more consistent with the sink/stream api, which would be more consistent with ```reunite```.

I've made that change and rebased to hopefully get a clean CI run.  Thanks for taking a look.